main.go

```diff
@@ -0,0 +1,174 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/codegangsta/cli"
+	"github.com/rancher/go-rancher-metadata/metadata"
+)
+
+const (
+	VERSION     = "0.0.1"
+	metadataURL = "http://rancher-metadata/2015-12-19"
+)
+
+func main() {
+	exit := func(err error) {
+		if err != nil {
+			log.Fatal(err)
+		}
+	}
+
+	app := cli.NewApp()
+	app.Author = "Rancher Labs"
+	app.EnableBashCompletion = true
+	app.Version = VERSION
+	app.Usage = `Hostname Update Service
+
+	Populates /etc/hosts of a rancher managed container based on currently registered
+	hosts in a given rancher environment`
+	app.Flags = []cli.Flag{
+		cli.IntFlag{
+			Name:  "update-interval",
+			Value: 5,
+			Usage: "time interval between refreshes of host list (in seconds)",
+		},
+	}
+	app.Action = func(c *cli.Context) {
+		exit(run(c))
+	}
+
+	exit(app.Run(os.Args))
+}
+
+func run(c *cli.Context) error {
+	metadataClient, err := metadata.NewClientAndWait(metadataURL)
+	if err != nil {
+		return err
+	}
+
+	interval := c.Int("update-interval")
+	rancherHosts := map[string]string{}
+
+	for {
+		err = repeat(metadataClient, rancherHosts)
+		if err != nil {
+			log.Errorf("Error updating /etc/hosts with hosts: [%v]", err)
+		}
+		<-time.After(time.Duration(time.Duration(interval) * time.Second))
+	}
+}
+
+func repeat(metadataClient *metadata.Client, rancherHosts map[string]string) error {
+	hosts, err := metadataClient.GetHosts()
+	if err != nil {
+		return err
+	}
+
+	changed := false
+
+	hostsMap := map[string]string{}
+
+	for _, host := range hosts {
+		if _, ok := hostsMap[host.Hostname]; ok {
+			// Do not add subsequent hosts with the
+			// duplicate hostnames
+			continue
+		}
+		if ip, ok := rancherHosts[host.Hostname]; !ok || ip != host.AgentIP {
+			// If the current host is not a part of the
+			// previous set of rancher hosts, then a new host
+			// was added
+			changed = true
+		}
+		hostsMap[host.Hostname] = host.AgentIP
+	}
+
+	// if a host was deleted, this will be true
+	if len(rancherHosts) != len(hostsMap) {
```

Comment:
Checking size doesn't seem sufficient.  Can you do proper full diff to see if the data changed.